milestones/tests/test_api.py

```diff
@@ -0,0 +1,652 @@
+# pylint: disable=invalid-name
+# pylint: disable=too-many-public-methods
+"""
+Milestones API Module Test Cases
+"""
+from opaque_keys.edx.keys import UsageKey
+
+import milestones.api as api
+import milestones.data as data
+import milestones.exceptions as exceptions
+import milestones.tests.utils as utils
+
+
+class MilestonesApiTestCase(utils.MilestonesTestCaseBase):
+    """
+    Main Test Case module for Milestones API
+    """
+    def setUp(self):
+        """
+        Milestones API Test Case scaffolding
+        """
+        super(MilestonesApiTestCase, self).setUp()
+        self.test_milestone = api.add_milestone({
+            'name': 'Test Milestone',
+            'namespace': unicode(self.test_course_key),
+            'description': 'Test Milestone Description',
+        })
+
+    def test_add_milestone(self):
+        """ Unit Test: test_add_milestone"""
+        milestone = api.add_milestone({
+            'name': 'Local Milestone',
+            'namespace': unicode(self.test_course_key),
+            'description': 'Local Milestone Description'
+        })
+        self.assertGreater(milestone['id'], 0)
+
+    def test_add_milestone_invalid_namespaces_throw_exceptions(self):
+        """ Unit Test: test_add_milestone_invalid_namespaces_throw_exceptions"""
+        try:
+            api.add_milestone({
+                'name': 'Local Milestone',
+                'namespace': '',  # Should throw an exception
+                'description': 'Local Milestone Description'
+            })
+            self.fail('Empty Milestone Namespace: Expected InvalidMilestoneException')  # pragma: no cover
```

Comment:
Thanks for the tip on using 'with' for each call -- I've changed the tests to this approach -- much cleaner